agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -1747,6 +1746,70 @@ impl DeriveSchema for UntaggedOperator {
             }
         }
 
+        fn get_numeric_operator_schema(
+            args: &[&Expression],
+            state: &mut ResultSetState,
+        ) -> Result<Schema> {
+            let mut type_set: BTreeSet<Schema> = set!();
+            let mut arg_schemas: BTreeSet<Schema> = set!();
+            // here we derive schema for all args and filter for numeric types only.
+            // we also use this time to capture if any args MAY be nullable, and to
+            // return null if any arg MUST be null or missing
+            for arg in args.iter() {
+                let arg_schema = arg.derive_schema(state)?;
+                match arg_schema.satisfies(&NULLISH.clone()) {
+                    Satisfaction::Must => return Ok(Schema::Atomic(Atomic::Null)),
+                    Satisfaction::May => {
+                        type_set.insert(Schema::Atomic(Atomic::Null));
+                    }
+                    Satisfaction::Not => {}
+                }
+                arg_schemas.insert(arg_schema.intersection(&NUMERIC.clone()));
+            }
+            // we can set the lower bound of numeric args as the largest type that is the minimum
+            // of one of the arguments. For example, if the lowest ordered type an argument can take
+            // on is a Double, the set that as the lower bound, as the operator will never return int or long.
+            let type_bound = arg_schemas
```

Comment:
```suggestion
            let lower_bound = arg_schemas
```
Let's use the word "lower" in the name